### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by AnkurChunekar**
## User Authentication
- User can Login, Signup and Logout.
- Preview: https://deploy-preview-3--velstream.netlify.app/
> Note: Homepage is not yet created.

### CODE REVIEW

File: src/helpers/utilityHelpers.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const checkIfAllInputsAreNotEmpty = (userData) => {
```
Comment (kart-c) [inline src/helpers/utilityHelpers.js:1]: maybe you can name this checkEmptyInputs? or inputValidation?

File: src/components/drawer/Drawer.jsx [hunk 2]
```diff
@@ -1,41 +1,64 @@
...
         </div>
-        </Link>
-        <Link to="/watchlater" >
+      </Link>
+      <Link to={ user ? "/watchlater" : "/login" }>
         <div className="drawer-item">
-            <span > <i className="fa-regular fa-clock"></i> </span>
-            <span> Watch Later </span>
+          <span>
+            {" "}
```
Comment (kart-c) [inline src/components/drawer/Drawer.jsx]: you can remove this empty braces if not required

File: src/components/drawer/Drawer.jsx [hunk 3]
```diff
@@ -1,41 +1,64 @@
...
-      <Link to="/" >
+export function Drawer() {
+  const { authState: { user } } = useAuth();
+
+  return (
+    <div className="nav-drawer dark fs-5">
+      <div className="drawer-item">
+        <span>
+          {" "}
+          <i className="fa-solid fa-house"></i>{" "}
```
Comment (RahullRawat) [inline src/components/drawer/Drawer.jsx]: If these curly braces are not necessary, you can remove them

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { createContext, useContext, useReducer } from "react";
+import { authReducer } from "../reducers";
+
+const defaultValue = { user: "", token: "" };
+
+const AuthContext = createContext(defaultValue);
+
+const AuthProvider = ({ children }) => {
+
+    const initialAuthState = { user: "", token: "" };
```
Comment (AtulPant2704) [inline src/context/auth-context.js]: No need of writing same initial state two times, you can just pass defaultValue only as your initialState in your reducer.

**Review by RahullRawat — APPROVED** [review]

**Review by kart-c — APPROVED** [review]
Auth is working as expected. Good job on the show password button on password inputs. Just a small suggestion, you can add a tooltip or the title attribute so the user knows what it is when hovered upon.

**Review by AtulPant2704 — APPROVED** [review]
Other than the one change mentioned, rest all looks good, dividing code into multiple files was really good, made it really easy to review.

**Pull request merged**